From 122fbafab2ff5533537caab0bb869e05002f8003 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Eil=C3=ADs=20N=C3=AD=20Fhlannag=C3=A1in?=
 <pidge@toganlabs.com>
Date: Thu, 26 Nov 2020 18:30:24 +0000
Subject: [PATCH 03/38] Copyright (C) 2014 Curt Brune
 <curt@cumulusnetworks.com>

SPDX-License-Identifier:     GPL-2.0
---
 arch/powerpc/kernel/pci-common.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/powerpc/kernel/pci-common.c b/arch/powerpc/kernel/pci-common.c
index 12659abf..6ce6a23e 100644
--- a/arch/powerpc/kernel/pci-common.c
+++ b/arch/powerpc/kernel/pci-common.c
@@ -1111,6 +1111,13 @@ void __devinit pcibios_setup_bus_devices(struct pci_bus *bus)
 			ppc_md.pci_dma_dev_setup(dev);
 
 		/* Read default IRQs and fixup if necessary */
+		if (pci_read_irq_line(dev) && (dev->devfn != 0)) {
+		     printk(KERN_WARNING
+			    "WARNING: Unable to map IRQ in device tree for pci device "
+			    "%04x:%02x:%02x.%d\n",
+			    pci_domain_nr(dev->bus), dev->bus->number, PCI_SLOT(dev->devfn),
+			    PCI_FUNC(dev->devfn));
+		}
 		pci_read_irq_line(dev);
 		if (ppc_md.pci_irq_fixup)
 			ppc_md.pci_irq_fixup(dev);
-- 
2.29.2

